core/src/poh_recorder.rs

```diff
@@ -105,6 +105,13 @@ impl PohRecorder {
         self.leader_schedule_cache.slot_leader_at(slot + 1, None)
     }
 
+    pub fn leader_after_slots(&self, slots: u64) -> Option<Pubkey> {
+        let slot =
+            leader_schedule_utils::tick_height_to_slot(self.ticks_per_slot, self.tick_height());
+        self.leader_schedule_cache
+            .slot_leader_at(slot + slots, None)
```

Comment:
passing None to the leader_schedule_cache might be a problem on epoch boundaries, because the leader schedule cache needs a bank in order to calculate the schedule for the next epoch. Maybe the poh_recorder should cache the latest bank that it saw so that it can use it here?

Reply:
Or, the caller of the function can handle it? The function is returning an Option. So the caller will know that the next leader is not known.

Reply:
The problem is, the caller (banking stage) can't do much with that information. It will just hold the txs instead of forwarding them to the correct leader.

In this case, the next leader at an epoch boundary is known, we just need a bank to use for calculating the leader schedule.